hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/BlockStreamFormat.java

```diff
@@ -38,13 +39,37 @@
  */
 public interface BlockStreamFormat {
 
+    /**
+     * Given the starting running hash and a stream of serialized {@link BlockItem}s, compute the new running hash by
+     * adding each {@link BlockItem}'s hash to the running hash one at a time.
+     *
+     * @param messageDigest the MessageDigest to use for computing the new running hash. This is an optimization to
+     *                      reduce the amount of garbage created when a caller calls this method in a loop.
+     * @param startRunningHash the starting current running hash of all record stream items up to this point in time.
+     * @param serializedItem the intermediary format serialized {@link BlockItem} to add to the running hash
+     * @return the new running hash, or startRunningHash if there were no items to add
+     */
+    @NonNull
+    Bytes computeNewRunningHash(
+            @NonNull final MessageDigest messageDigest,
+            @NonNull final Bytes startRunningHash,
+            @NonNull final Bytes serializedItem);
+
+    /**
+     * Returns an instance of a non-thread safe MessageDigest that can be used to compute the hash of block items.
```

Comment:
We want to be very careful with a non thread safe object if we're using `ConcurrentBlockStreamWriter`